Content.Server/Gateway/Systems/GatewayGeneratorSystem.cs

```diff
@@ -0,0 +1,222 @@
+using System.Linq;
+using System.Numerics;
+using Content.Server.Gateway.Components;
+using Content.Server.Parallax;
+using Content.Server.Procedural;
+using Content.Shared.Dataset;
+using Content.Shared.Movement.Components;
+using Content.Shared.Parallax.Biomes;
+using Content.Shared.Physics;
+using Content.Shared.Procedural;
+using Content.Shared.Salvage;
+using Robust.Shared.Map;
+using Robust.Shared.Map.Components;
+using Robust.Shared.Physics.Collision.Shapes;
+using Robust.Shared.Physics.Components;
+using Robust.Shared.Physics.Systems;
+using Robust.Shared.Prototypes;
+using Robust.Shared.Random;
+using Robust.Shared.Timing;
+using Robust.Shared.Utility;
+
+namespace Content.Server.Gateway.Systems;
+
+/// <summary>
+/// Generates gateway destinations regularly and indefinitely that can be chosen from.
+/// </summary>
+public sealed class GatewayGeneratorSystem : EntitySystem
+{
+    [Dependency] private readonly IGameTiming _timing = default!;
+    [Dependency] private readonly IMapManager _mapManager = default!;
+    [Dependency] private readonly IPrototypeManager _protoManager = default!;
+    [Dependency] private readonly IRobustRandom _random = default!;
+    [Dependency] private readonly ITileDefinitionManager _tileDefManager = default!;
+    [Dependency] private readonly BiomeSystem _biome = default!;
+    [Dependency] private readonly DungeonSystem _dungeon = default!;
+    [Dependency] private readonly FixtureSystem _fixtures = default!;
+    [Dependency] private readonly GatewaySystem _gateway = default!;
+    [Dependency] private readonly MetaDataSystem _metadata = default!;
+    [Dependency] private readonly SharedMapSystem _maps = default!;
+    [Dependency] private readonly SharedPhysicsSystem _physics = default!;
+
+    [ValidatePrototypeId<DatasetPrototype>]
+    private const string PlanetNames = "names_borer";
+
+    // TODO:
+    // Fix shader some more
+    // Show these in UI
+    // Use regular mobs for thingo.
+
+    // Use salvage mission params
+    // Add the funny song
+    // Put salvage params in the UI
+
+    // Re-use salvage config stuff for the RNG
+    // Have it in the UI like expeditions.
+
+    // Also add weather coz it's funny.
+
+    // Add songs (incl. the downloaded one) to the ambient music playlist for planet probably.
+    // Copy most of salvage mission spawner
+
```

Comment:
This is a fat TODO but I mainly want to make sure people like the idea before killing expeds and moving it over here.